src/useConstantCallback.js

```diff
@@ -0,0 +1,30 @@
+// @flow
+import * as React from 'react'
+
+/**
+ * Creates a callback, with dependencies, that will be
+ * instance === for the lifetime of the component.
+ */
+export default function useConstantCallback(callback, deps) {
```

Comment:
Would something like the following help? It will always return the same instance for === purposes, but will call
the latest function passed in. No need to care about callback dependencies at all:

```
export function useWrappedCallback(callback) {
	const ref = React.useRef(callback);
	React.useEffect(() => { ref.current = callback; });
	return React.useCallback((...args) => ref.current.apply(null, args), []);
}
```

Reply:
Ooh!! I like this a lot!

Reply:
This is epic. I had to stare at it for some time tho. So basically we always return the same identity which is inline with earlier version. BUT we work with inner function which is passed from render which is up to date on every render, thus no need for deps at all indeed and also this scales perfectly, i.e. over time when new vars are added or removed we have to change NOTHING! If I knew about this smart pattern I'd use this instead in previous PR. So I am all for it.

Reply:
The only thing I'd TRY to maybe improve is the name to provide as much info on what this actually does explicit as possible like `useFixedWrapperCallback` or `useCallbackWithFixedWrapper` or `useFixedProxy`. (cc: @erikras)

Reply:
Updated PR with your implementation. MUCH nicer!